tests/datatype.js

```diff
@@ -23,7 +23,7 @@ test('datatype - create, encode, decode', async (t) => {
   })
 
   const updated = await dataType.update(
-    Object.assign({}, created, { title: 'hi' })
+    Object.assign({}, created, { title: 'hi', links: [created.id] })
```

Comment:
This should be `created.version`. I wonder if we should validate if a given version actually exists? It wouldn't actually require a read, we could just parse the hypercore key and index from the version and check that hypercore has length > index. Although perhaps maybe also worth validating the record that is being linked to check it is of the same data type? To avoid really weird data.

Reply:
:+1: fixed the test type and added validation that checks that links exist and are of the same datatype.